### PR TITLE
Fix CMake syntax to enable memory analyzer tests

### PR DESCRIPTION
**Pull request by tautschnig**
"option" requires the help text as second, not as third argument. This
resulted in memory analyzer tests being disabled no matter what the OS
being built on.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: unit/memory-analyzer/gdb_api.cpp [hunk 1]
```diff
@@ -18,13 +18,61 @@ Author: Malte Mues <mail.mues@gmail.com>
...
+          "\n"
+          "  return 0;\n"
+          "}\n";
+    of.close();
+
+    REQUIRE(run("gcc", {"gcc", "-g", "-o", compiled(), tmp()}) == 0);
+  }
+
+  temporary_filet compiled;
+};
```
Comment (TGWDB) [inline unit/memory-analyzer/gdb_api.cpp]: This seems unnecessarily ugly, is there a reason to do it this way?

File: unit/memory-analyzer/gdb_api.cpp [hunk 2]
```diff
@@ -67,9 +115,34 @@ void gdb_api_internals_test()
...
+          "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+          "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+          "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+          "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+          "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+          "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+          "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+          "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa\n"
+          "xyz";
+    of.close();
```
Comment (TGWDB) [inline unit/memory-analyzer/gdb_api.cpp]: Again, is this the best way to do this?

File: unit/memory-analyzer/gdb_api.cpp (hunk 1 above)
Comment (tautschnig) [reply unit/memory-analyzer/gdb_api.cpp]: If, as was the case before, we keep the contents in a separate file, then this file must either be included in the compiled unit test binary (e.g., via `#include`), or else the compiled unit test binary necessarily relies on the source tree being present as well. It seems reasonable to assume that the source tree will be present, but then also the exact location of the file must be known.

I made these changes as I found the above conditions not to be satisfied in my ad-hoc testing for I wasn't aware where the unit test had to be run from to make things work. That being said, we have the same problem in other unit tests. So maybe it's not all that important to fix if people are concerned about this approach.

Comment (TGWDB) [reply unit/memory-analyzer/gdb_api.cpp]: I understand the motivation, I was more considering that there might be a prettier way to have the text inside the file.

**Review by TGWDB — APPROVED** [review]

File: unit/memory-analyzer/gdb_api.cpp (hunk 2 above)
Comment (tautschnig) [reply unit/memory-analyzer/gdb_api.cpp]: I've modified this to use `ansi-c/file_converter` and `#include` them instead.

**Review by TGWDB — APPROVED** [review]

**Review by chrisr-diffblue — APPROVED** [review]

**Pull request merged**